extensions/guacamole-auth-totp/src/main/java/org/apache/guacamole/auth/totp/user/TOTPUser.java

```diff
@@ -82,13 +106,15 @@ public User getUndecorated() {
     public void setAttributes(Map<String, String> attributes) {
 
         // Create independent, mutable copy of attributes
-        attributes = new HashMap<String, String>(attributes);
-
-        // Do not expose any TOTP-related attributes outside this extension
-        attributes.remove(TOTP_KEY_SECRET_ATTRIBUTE_NAME);
-        attributes.remove(TOTP_KEY_CONFIRMED_ATTRIBUTE_NAME);
```

Comment:
Is there a use case for allowing the user to reconfirm their existing key, rather than requiring that an entirely new key be generated in order to reset?

Reply:
I think this was mentioned in the mailing list somewhere along the way.  I don't have any particular affinity for it - I think *most* of the situations where you'd want to un-confirm you it you'd also want to reset it, but I left the option in there.  Happy to go either direction.